### PR TITLE
[WFLY-6341] Testcase for basic use of BouncyCastleProvider

### PR DESCRIPTION
**Pull request by jbliznak**
https://issues.jboss.org/browse/WFLY-6341

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/bc/BouncyCastleModuleTestCase.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                new RuntimePermission("getProtectionDomain")
+        ), "permissions.xml");
+        archive.setManifest(new StringAsset(""
+                + "Manifest-Version: 1.0\n"
+                + "Dependencies: org.bouncycastle\n"));
+        return archive;
+    }
+
+    @Test
+    @OperateOnDeployment(BC_DEPLOYMENT)
```
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/bc/BouncyCastleModuleTestCase.java]: no need

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/bc/BouncyCastleModuleTestCase.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    public void testBouncyCastleProviderIsUsableThroughJceApi() throws Exception {
+
+        BouncyCastleProvider bcProvider = null;
+        try {
+            bcProvider = new BouncyCastleProvider();
+            useBouncyCastleProviderThroughJceApi(bcProvider);
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/bc/BouncyCastleModuleTestCase.java]: remove

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/bc/BouncyCastleModuleTestCase.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        // Encrypt the text
+        byte[] textEncrypted = desCipher.doFinal(text);
+        System.out.println("Text Encryted [Byte Format]: " + textEncrypted);
+
+        // Initialize the same cipher for decryption
+        desCipher.init(Cipher.DECRYPT_MODE, myDesKey);
+
+        // Decrypt the text
+        byte[] textDecrypted = desCipher.doFinal(textEncrypted);
+        System.out.println("Text Decryted: " + new String(textDecrypted));
```
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/bc/BouncyCastleModuleTestCase.java]: remove all system.outs from this method or replace them with logger trace/debug calls.

**Comment by jbliznak** [comment]
@ctomc, thanks for review, notes incorporated.

**Pull request merged**